pkg/archive/archive.go

```diff
@@ -399,7 +399,9 @@ func ReadSecurityXattrToTarHeader(path string, hdr *tar.Header) error {
 			return errors.Wrapf(err, "failed to read %q attribute from %q", xattr, path)
 		}
 		if capability != nil {
-			hdr.Xattrs = make(map[string]string)
+			if hdr.Xattrs == nil {
+				hdr.Xattrs = make(map[string]string)
+			}
```

Comment:
Can this be moved out of the for loop before L396?

Reply:
ok